client/src/driver/BitbakeESDK.ts

```diff
@@ -85,3 +86,27 @@ export function generateTasksDefinitions (workspace: DevtoolWorkspaceInfo, bitba
   logger.info(`Generated ${vscodeTasksPath} for ${recipeName}`)
   logger.debug(`Tasks: ${JSON.stringify(vscodeTasks)}`)
 }
+
+export async function generateCPPProperties (workspace: DevtoolWorkspaceInfo, bitBakeProjectScanner: BitBakeProjectScanner, client: LanguageClient): Promise<void> {
+  const vscodeCppPropertiesPath = path.join(workspace.path, '.vscode', 'c_cpp_properties.json')
+  const vscodeCppProperties = loadJsonFile(vscodeCppPropertiesPath)
+  const recipe = workspace.name
+
+  const STAGING_BINDIR_TOOLCHAIN = await bitBakeProjectScanner.resolveContainerPath(await getVariableValue(client, 'STAGING_BINDIR_TOOLCHAIN', recipe, true), true) ?? ''
+  const CXX = await getVariableValue(client, 'CXX', recipe, true)
```

Comment:
The `getVariableValue` is declared in the `languageClient.ts`, I think we could make it a property of `BitbakeRecipeScanner.ts`. It also has a reference to the initialized client. We don't need to pass the client reference across multiple files, just import the `bitbakeRecipeScanner` where we need it. It should be beneficial since we are calling this function more.

Reply:
It makes sense but I also don't like singletons much in the client. It's cumbersome when testing and not a great practice in OOP. I can consider a refactoring, but it could be in a separate PR.

Reply:
sure, I just feel we should try not passing the `connection` and `client` unless we have to. Especially when they are passed across multiple layers/files.